indico/cli/i18n.py

```diff
@@ -113,79 +248,261 @@ def _wrapper(**kwargs):
+
+
+@cli.command(short_help='Push .pot files to Transifex.')
+def push():
+    """Push .pot files to Transifex."""
+    try:
+        subprocess.run(['tx', 'push', '-s'], check=True)
+    except subprocess.CalledProcessError:
+        click.echo(click.style('Error pushing to transifex ', fg='red', bold=True), err=True)
+
+
+@cli.command(short_help='Pull translated .po files from Transifex.')
+@click.argument('languages', nargs=-1, required=False)
+def pull(languages):
+    """Pull translations from Transifex.
+
+    Pulls only the language codes specified in `languages` if present.
+    If no `languages`, it pulls all the languages currently available in the translations folder.
+    To pull a new language, simply pass the new language code in the argument.
+    """
+    if not languages:
+        languages = [d for d in os.listdir(TRANSLATIONS_DIR) if os.path.isdir(os.path.join(TRANSLATIONS_DIR, d))]
+    try:
+        for language in languages:
+            subprocess.run(['tx', 'pull', '-l', language, '-f'], check=True)
+            click.echo(f'Translations updated for {language}.')
+    except subprocess.CalledProcessError:
+        click.echo(click.style('Error pulling from transifex', fg='red', bold=True), err=True)
+    finally:
+        language_renames = {'zh_CN.GB2312': 'zh_Hans_CN'}  # other lang codes requiring rename should be added here
+        languages = set(languages) & set(language_renames.keys())
+        for code in languages:
+            if os.path.exists(os.path.join(TRANSLATIONS_DIR, code)):
+                shutil.rmtree(os.path.join(TRANSLATIONS_DIR, language_renames[code]))
+                shutil.move(os.path.join(TRANSLATIONS_DIR, code),
+                            os.path.join(TRANSLATIONS_DIR, language_renames[code]))
 
 
 @cli.command()
-def check_format_strings():
+def check_format_strings(root_path='indico/translations'):
     """Check whether format strings match.
 
     This helps finding cases where e.g. the original string uses
     ``{error}`` but the translation uses ``{erro}``, resulting
     in errors when using the translated string.
     """
-    root_path = 'indico/translations'
     paths = set()
     for root, dirs, files in os.walk(root_path):
         for file in files:
             if file.endswith('.po'):
                 paths.add(os.path.join(root, file))
-    all_valid = True
     for path in paths:
         invalid = _get_invalid_po_format_strings(path)
         if invalid:
-            all_valid = False
             click.echo(f'Found invalid format strings in {os.path.relpath(path, root_path)}')
             for item in invalid:
                 click.echo(cformat('%{yellow}{}%{reset} | %{yellow!}{}%{reset}\n%{red}{}%{reset} != %{red!}{}%{reset}')
                            .format(item['orig'], item['trans'],
                                    list(item['orig_placeholders']), list(item['trans_placeholders'])))
             click.echo()
-    sys.exit(0 if all_valid else 1)
```

Comment:
This logic needs to stay since we use it in CI!

Reply:
Eventually I moved the logic into a different function, leaving this command to exit properly.  That way I can check the format strings before compiling and it would not just exit the program on the first fail when checking for multiple plugins.